src/main/java/org/icatproject/ids/thread/DfRestorer.java

```diff
@@ -45,46 +45,51 @@ public DfRestorer(List<DfInfo> dfInfos, PropertyHandler propertyHandler, FiniteS
 	 */
 	@Override
 	public void run() {
-		/*
-		 * This code avoids unnecessary calls to restore files. It will not
-		 * generally remove anything from the list of files to restore as
-		 * pointless restores are normally filtered out earlier.
-		 */
-		Iterator<DfInfo> iter = dfInfos.iterator();
-		while (iter.hasNext()) {
-			DfInfo dfInfo = iter.next();
-			try {
-				if (mainStorageInterface.exists(dfInfo.getDfLocation())) {
-					iter.remove();
-					fsm.removeFromChanging(dfInfo);
+		try {
+			/*
+			 * This code avoids unnecessary calls to restore files. It will not
+			 * generally remove anything from the list of files to restore as
+			 * pointless restores are normally filtered out earlier.
+			 */
+			Iterator<DfInfo> iter = dfInfos.iterator();
+			while (iter.hasNext()) {
+				DfInfo dfInfo = iter.next();
+				try {
+					if (mainStorageInterface.exists(dfInfo.getDfLocation())) {
+						iter.remove();
+						fsm.removeFromChanging(dfInfo);
+					}
+				} catch (IOException e) {
+					logger.error("Check on existence of {} failed with {} {}", dfInfo.getDfLocation(), e.getClass(),
+							e.getMessage());
 				}
-			} catch (IOException e) {
-				logger.error("Check on existence of {} failed with {} {}", dfInfo.getDfLocation(), e.getClass(),
-						e.getMessage());
 			}
-		}
 
-		try {
 			Set<DfInfo> failures = archiveStorageInterface.restore(mainStorageInterface, dfInfos);
 			for (DfInfo dfInfo : dfInfos) {
-				if (failures.contains(dfInfo)) {
-					fsm.recordFailure(dfInfo.getDfId());
-					logger.error("Restore of " + dfInfo + " failed");
-				} else {
-					fsm.recordSuccess(dfInfo.getDfId());
-					logger.debug("Restore of " + dfInfo + " completed");
+				try {
```

Comment:
I'd say this inner try ... catch ... finally clause is not needed. It is essentially redundant with the outer one: the outer finally already guarantees removing all dfInfos from changing.

Reply:
The inner try..catch.finally clause is purely for consistency with the other Runnable classes. I agree that it is not needed and can be removed.